feat: Make the Portfolio Homepage more accessible

gvgesanto2 opened the pull request:
## Description

**Task:** https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/6_accessability.md

This PR checks all the items listed in the article [Easy Checks – A First Review of Web Accessibility](https://www.w3.org/WAI/test-evaluate/preliminary/) in order to make the Portfolio Homepage more accessible. The items checked and their respective changes are the following:
-  Page titles: 
   - [Update the page title for a more descriptive one](https://github.com/gvgesanto2/microverse-portfolio-mobile/compare/accessibility?expand=1#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R14)
- Image text alternatives:
   - [Add text alternatives for the social linked icons using the ``aria-label`` attribute](https://github.com/gvgesanto2/microverse-portfolio-mobile/compare/accessibility?expand=1#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R60-R112)
   -  [Add a more descriptive ``alt`` attribute for the works section projects images](https://github.com/gvgesanto2/microverse-portfolio-mobile/compare/accessibility?expand=1#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R124-R238)
   - [Shorten the text alternative for the languages' logo images in the about-me section](https://github.com/gvgesanto2/microverse-portfolio-mobile/compare/accessibility?expand=1#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051L340-R413)
- Text headings: no issues found
- Color contrast: no issues found
- Resize: no issues found
- Interaction: 
  - [Add back the browser's default focus styles for buttons](https://github.com/gvgesanto2/microverse-portfolio-mobile/compare/accessibility?expand=1#diff-53d336f74a77d894988ebd7dd72165e06fe72825f4128bbdbd7ce3626e70267eL53)
  - [Set a focus style for the contact form inputs and textarea](https://github.com/gvgesanto2/microverse-portfolio-mobile/compare/accessibility?expand=1#diff-ab3f9e9705f27007f964c4ef6bed651bfb21f34c89eb56fb06ad34062e2b8416R24-R26)
 - Moving content: no issues found
 - Multimedia: no issues found
 - The basic structure of the page: no issues found

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented my code
- [X] I have deleted any unused code as comments in my codebase

gvgesanto2 committed:
feat: make the page more accessible

gvgesanto2 committed:
feat: Add text alternatives for the social linked icons in the about-me section

zieeco commented on css/base/base.css:
```diff
@@ -50,7 +50,6 @@ button {
   background-color: transparent;
-  outline: none;
   cursor: pointer;
 }
 
```
- [ ] Hey, Great job on your accessibility, however there are some issues you still need to address. I'd pointing them out below, Please kindly look them up in your file and fix them. I just used this inline comment to be able make the review..

- [ ] Hey, Please kindly add the label tag to your form input and textarea.  A form should have the label tag as the label tag point or specify a label for an <input>, <select>, <textarea> element of a form to improve accessibility for screen users. There are many ways to add the label tag. use this for yours

`<label for="username"><input type="text" name="username" id="username" maxlength="30" placeholder="username" required></label>`

zieeco replied:
- [ ] please, kindly fix this color contrast issue.
- Use this [chrome extetnsion](https://chrome.google.com/webstore/detail/wcag-color-contrast-check/plnahcmalebffmaghcpcmpaciebdhgdf?hl=en) to analyze and fix all your color contrast issue.
- Your `LET'S CONNECT`.
- Your contact section background color.
- Your list items e.g: `html css javascript`.

gvgesanto2 replied:
## Hi @zieeco,

Thank you for your review and feedback! However, I got a question about this required change:

> * [ ]  Hey, Please kindly add the label tag to your form input and textarea.  A form should have the label tag as the label tag point or specify a label for an , , <textarea> element of a form to improve accessibility for screen users. There are many ways to add the label tag. use this for yours
>   
>   
>   <label for="username"><input type="text" name="username" id="username" maxlength="30" placeholder="username" required></label>

If I add a ``label`` tag in the way that was suggested, I will change the original design proposed by the [Figma Template 1](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1). I wouldn't like to do that because one of the core project requirements is to stick to this template design as much as possible. 

So, I thought of another solution that will achieve accessibility without changing the original design, which is to add an ``aria-label`` attribute to the form inputs and textarea. This technique is used for example in the input of the login form of Google (which doesn't have a ``label`` tag associated with it but an ``aria-label`` attribute instead):

![google-login-form](https://user-images.githubusercontent.com/64566209/166605641-856a2920-ef5b-44e1-81fd-401aa78c55d9.png)

This ``aria-label`` will provide a label for the input and textarea objects that can be read by assistive technology (e.g. screen readers) without changing my portfolio design. **So, what do you think? Can I proceed with the change described in this comment instead of the one that was suggested?**

zieeco reviewed (changes requested):
Hi @Team


Great work on improving your portfolio accessibility 👏 
You've done well implementing them, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._


<img src="https://media0.giphy.com/media/8UGVQESpDMMQl7yd3F/giphy.gif"/>


## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gvgesanto2 commented:
## Hi @zieeco,

I also have a question about this required change:

> * [ ]  please, kindly fix this color contrast issue.
> * Use this [chrome extetnsion](https://chrome.google.com/webstore/detail/wcag-color-contrast-check/plnahcmalebffmaghcpcmpaciebdhgdf?hl=en) to analyze and fix all your color contrast issue.
> * Your `LET'S CONNECT`.
> * Your contact section background color.
> * Your list items e.g: `html css javascript`.

I have downloaded the Chrome extension you suggested, but unfortunately, I face the same problem mentioned in [my last comment](https://github.com/gvgesanto2/microverse-portfolio-mobile/pull/9#discussion_r864356289). To solve the color contrast issues it'd be necessary to make changes in the original [Figma design](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1). 

That is against one of the core requirements of the project:
![design-requirement](https://user-images.githubusercontent.com/64566209/166615905-6ae4721c-f549-43a8-819b-6317584d0f9b.png)
_click on the image above for seeing it in a better resolution_

This rule explicitly tells us to keep the colors used in [this Figma template](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1). So, as I've used the exact same colors used on that template, **I think that if I change it now, that could be pointed as an error by another code reviewer in future pull requests**.

Besides that, as a front-end developer, once I receive a design made by a UI/UX engineer I might not be allowed to change it. If I find some color contrast problems related to it, for example, I believe I probably should communicate those problems to the UI/UX engineer responsible for the design to address them, because he's the one who's going to be able to find the best colors to solve the color contrast issues while keeping the design "look-and-feel" and consistency intact.

**That said, I'd like to ask your permission to skip this change you suggested.**

gvgesanto2 committed:
feat: Add an aria-label attribute for the contact form inputs and textarea

gvgesanto2 commented:
## Description

**Task:** https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/6_accessability.md

This PR addresses the following changes requested by the [last review](https://github.com/gvgesanto2/microverse-portfolio-mobile/pull/9#pullrequestreview-961159736):

> * [X]  Hey, Please kindly add the label tag to your form input and textarea.  A form should have the label tag as the label tag point or specify a label for an , , <textarea> element of a form to improve accessibility for screen users. There are many ways to add the label tag. use this for yours
>   
>   
>   <label for="username"><input type="text" name="username" id="username" maxlength="30" placeholder="username" required></label>

**Solution adopted:** It was used an ``aria-label`` attribute in the contact form inputs and textarea in order to create a label that can be read by assistive technology (e.g. screen readers) without changing the [original project design](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1), as described in [this comment](https://github.com/gvgesanto2/microverse-portfolio-mobile/pull/9#discussion_r864356289).

> * [ ]  please, kindly fix this color contrast issue.
> * Use this [chrome extetnsion](https://chrome.google.com/webstore/detail/wcag-color-contrast-check/plnahcmalebffmaghcpcmpaciebdhgdf?hl=en) to analyze and fix all your color contrast issue.
> * Your `LET'S CONNECT`.
> * Your contact section background color.
> * Your list items e.g: `html css javascript`.

**Solution adopted:** This change was skipped by this PR because of the reasons described in [my last comment](https://github.com/gvgesanto2/microverse-portfolio-mobile/pull/9#issuecomment-1116895105).
Besides that, according to this explanation from the [Easy Checks – A First Review of Web Accessibility](https://www.w3.org/WAI/test-evaluate/preliminary/) article:

![color-contrast](https://user-images.githubusercontent.com/64566209/166719621-a5bb0683-0f1f-4213-ab7f-0b50d3be7094.png)

The high contrast suggested by the [last review](https://github.com/gvgesanto2/microverse-portfolio-mobile/pull/9#pullrequestreview-961159736) may not be proper for people with reading disabilities such as dyslexia, who need low luminance. So given the fact that some people need high contrast and some need low luminance, I believe that the best solution for this accessibility issue might be the one highlighted by the red box in the text snippet above, which is that the web pages should work properly when the user uses some tool for changing its text and background colors. 
**That said, this page works properly when its text or background colors are changed by some tool used by the users.**

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented my code
- [X] I have deleted any unused code as comments in my codebase

OmarRamoun reviewed (approved):
# 🟢 Project Approved! 🏆

## Hi @gvgesanto2 👋

Your project is complete! ✔️ 🏁

Finally, **you did it**! 🎉 👏. Your code was very professional 💯 and I can see you have a great chance to be a Great Software Engineer ⛑️.

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉. And Welcome to Microverse 👋.

> **Note:** You can now take a break even if short before you move on to the next project. Happy 😄 Coding 🖥️! and Have a nice day 🌞.

## Highlights 🌟

- ✔️ implemented previous code review requested changes
- ✔️ descriptive commit messages
- ✔️ descriptive PR title and description
- The code is top notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

## Thank you for your time 🙏 and call me if you need help or clarification ☎️
- This is My **GitHub**: [![@mrramoun](https://img.shields.io/github/followers/MrRamoun?label=Ramoun&style=social)](https://github.com/mrramoun)
- This is My **Twitter**: [![@ramoun16](https://img.shields.io/twitter/follow/ramoun16?label=ramoun16&style=social)](https://twitter.com/ramoun16)
- This is My **Slack**: [![Ramoun](https://img.shields.io/badge/slack-@Ramoun-brightgreen)](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
